### DISCUSSION TITLE
Scope specific month to specific month

### DISCUSSION THREAD

**Discussion by bobbybosler** [liedekef/events-made-easy#349] [General]
Hey Franky,

Love your stuff. Consider it the single best events plugin out there. Hands down. I use it for every site I build, just about.

I was wondering if you have the option of starting an events list at a certain month and ending in a specific month in a way that automatically changes after a certain date to those months for next year. For instance, starting `this_January` and ending `this_May` or something like that. I know I can "hard code" `2024-01-01--2024-05-01`, but what I'd really like is to be able to set it and not have to change it and have things "rollover" at a particular date. 

Let me give a little more context. On thewar.team/#schedule, I've got an itinerary for a tour I conduct. My meetings are always from January-May each year—that constitutes a "tour." I am using the `[events_list]` shortcode, and I currently have the `scope` set to `-4m--8m`.  What I really want is, say, for this upcoming itinerary, for the next tour to appear from September 2023 through August 2024. But I don't want to have to change it, and I don't want "tours" to get mixed up.

I'm not exactly sure how that would be done. Perhaps a `this_year`, but with an offset of -4m or something like that? I just thought I'd reach out to see if you had any ideas whether this is possible with the current parameters or whether it would require something new. Thanks for considering!!

**Comment by liedekef** [comment]
Sorry, but the current scope filters won't be able to do what you want here. Either try a different approach or use the available EME filters.

**Reply by liedekef** [reply]
I've looked a little bit further and something like this might be of help:

`scope=0000-01--relative+4m`
I'm trying to code this up now

**Reply by bobbybosler** [reply]
Awesome. I've implemented it. So, help me out here. What is the "0000" for? Also, in the example you gave `scope=0000-01--relative+4m`, so that will wait until this coming January and will display all events from that time until 4 months in the future? If so, what happens when February 2024 rolls around? Will it switch to the next January+4m or will it continue until those 4m have elapsed and then switch to the next January+4m? Just wanting to make sure I understand to prevent surprises. 

You're the best, Franky!

**Reply by liedekef** [reply]
0000 means the year is irrelevant (it's currently for the regex-matching too :-) ). 
`scope=0000-01--relative+4m` should (for all intents and purposes) show all events starting in that range (the beginning of January until the end of the month 4 months later, so until the end of May).
After the end of May it will show the events of the following year Jan-May.
